src/nvim/os/shell.c

```diff
@@ -166,17 +168,16 @@ int os_call_shell(char_u *cmd, ShellOpts opts, char_u *extra_args)
 ///             returned buffer is not NULL)
 /// @return the return code of the process, -1 if the process couldn't be
 ///         started properly
-int os_system(const char *cmd,
+int os_system(char **argv,
               const char *input,
               size_t len,
               char **output,
               size_t *nread) FUNC_ATTR_NONNULL_ARG(1)
 {
-  return shell(cmd, NULL, input, len, output, nread, true, false);
+  return shell(argv, input, len, output, nread, true, false);
 }
 
-static int shell(const char *cmd,
-                 const char *extra_args,
+static int shell(char **argv,
```

Comment:
I don't think this function should be called `shell` as it doesn't necessarily use the shell anymore. perhaps `call_argv` or `call_system` or something like that.